src/u2fprotocol.rs

```diff
@@ -49,18 +49,8 @@ where
     register_data.extend(application);
 
     let flags = U2F_REQUEST_USER_PRESENCE;
-    let register_resp = send_apdu(dev, U2F_REGISTER, flags, &register_data)?;
-
-    if register_resp.len() < 2 {
-        return Err(io_err("unexpected response"));
-    }
-
-    if register_resp.len() > 2 {
-        // Real data, we're done
-        return Ok(register_resp);
-    }
-
-    status_word_to_result(register_resp[0], register_resp[1], vec![])
+    let (resp, status) = send_apdu(dev, U2F_REGISTER, flags, &register_data)?;
+    status_word_to_result(status[0], status[1], resp)
```

Comment:
Nit: It feels weird to pass `status[0], status[1]` repeatedly when we could be passing `status` and asserting that it's exactly 2 bytes.

Reply:
Good point, fixed with the latest commit.